src/allmydata/frontends/magic_folder.py

```diff
@@ -153,21 +258,38 @@ def startService(self):
         self._stats_provider.count('magic_folder.dirs_monitored', 1)
         return d
 
-    def upload_ready(self):
-        """upload_ready is used to signal us to start
-        processing the upload items...
+    def ready(self):
+        """ready is used to signal us to start
+        processing the upload and download items...
         """
-        self.is_upload_ready = True
-        self._turn_deque()
+        self.is_ready = True
+        self._turn_upload_deque()
+        self._turn_download_deque()
+
+    def _turn_download_deque(self):
+        if self._stopped:
+            return
+        try:
+            file_path, file_node, metadata = self._download_deque.pop()
+        except IndexError:
+            self._log("magic folder upload deque is now empty")
+            self._download_lazy_tail = defer.succeed(None)
+            self._download_lazy_tail.addCallback(lambda ign: task.deferLater(reactor, self._remote_scan_delay, self._scan_remote_collective))
+            self._download_lazy_tail.addCallback(lambda ign: task.deferLater(reactor, 0, self._turn_download_deque))
+            return
+        self._download_lazy_tail.addCallback(lambda ign: task.deferLater(reactor, 0, self._download_file, file_path, file_node))
+        self._download_lazy_tail.addCallback(lambda ign: task.deferLater(reactor, self._remote_scan_delay, self._turn_download_deque))
 
-    def _append_to_deque(self, path):
+    def _append_to_upload_deque(self, path):
+        if path in self._upload_pending:
+            return
```

Comment:
This is correct when `_append_to_upload_deque` is called from `_notify`, but is it correct when called from `_scan`?